### ISSUE TITLE
Incompatible with REI 1.17

### ISSUE THREAD

**Issue by shedaniel** [Queerbric/Inspecio#8]
Custom TooltipComponents were not handled in the best way. (see TooltipComponent#of), You may want to mixin that to return a proper value for REI (/or other mods) to render that easier. (Instead of mixing into the lambda in the screen)

https://gist.githubusercontent.com/natanfudge/aa81ecfcdf0c94cf7ff777a1cd7d622a/raw/19ad3a6e99db40a263a32eac9f517f4cd51a7d66/crash.txt

**Comment by LambdAurora** [comment]
Welcome to Mixin!

For context TooltipComponent is an interface. Mixins targeting interfaces must be interfaces as well.

Now here's the tricky part, we need a static Injector to `TooltipComponent#of`, so let's try with this:
```java
@Mixin(TooltipComponent.class)
public interface TooltipComponentMixin {
	@Inject(method = "of(Lnet/minecraft/client/item/TooltipData;)Lnet/minecraft/client/gui/tooltip/TooltipComponent;", at = @At("HEAD"), cancellable = true)
	private static void onOf(TooltipData data, CallbackInfoReturnable<TooltipComponent> cir) {
		if (data instanceof ConvertibleTooltipData convertible) {
			cir.setReturnValue(convertible.getComponent());
		}
	}
}
```

Game starts, and Mixin will yell this and refuse to apply it:
```
[21:00:14] [main/ERROR] (mixin) inspecio.mixins.json:TooltipComponentMixin: Interface mixin contains a non-public method! Found onOf(Lnet/minecraft/client/item/TooltipData;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfoReturnable;)V in inspecio.mixins.json:TooltipComponentMixin
 org.spongepowered.asm.mixin.transformer.throwables.InvalidInterfaceMixinException: Interface mixin contains a non-public method! Found onOf(Lnet/minecraft/client/item/TooltipData;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfoReturnable;)V in inspecio.mixins.json:TooltipComponentMixin
 ```
 
 Uh, weird isn't it?
 Let's try again with public then!
 
 ```
 Caused by: org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: Mixin inspecio.mixins.json:TooltipComponentMixin contains non-private static method onOf(Lnet/minecraft/client/item/TooltipData;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfoReturnable;)V [ -> MAIN Applicator Phase -> inspecio.mixins.json:TooltipComponentMixin -> Apply Methods -> (Lnet/minecraft/client/item/TooltipData;Lorg/spongepowered/asm/mixin/injection/callback/CallbackInfoReturnable;)V:handler$zza000$onOf]
 ```
 
So... uh.... There's a big issue there.
 
This cannot be fixed without a fix in upstream (SpongePowered/Mixin#497), or a fix in Fabric's Mixin.

**Comment by shedaniel** [comment]
Thanks for the info, I didn’t notice that it was an interface, will fix this on my side.
